integration.test.js

```diff
@@ -0,0 +1,116 @@
+/* global describe, beforeAll, afterAll, it, expect */
+require('dotenv').config()
+const Fastify = require('fastify')
+const fetch = require('cross-fetch')
+
+if (
+  !process.env.AUTH0_DOMAIN ||
+  !process.env.AUTH0_CLIENT_ID ||
+  !process.env.AUTH0_CLIENT_SECRET ||
+  !process.env.AUTH0_API_AUDIENCE
+) {
+  throw new Error('Integration tests needs a set of environment variables to be set')
+}
+
+async function buildServer() {
+  const fastify = Fastify()
+
+  // Setup fastify-auth0-verify
+  async function authenticate(fastify) {
+    fastify.register(require('./index.js'), {
+      domain: process.env.AUTH0_DOMAIN,
+      secret: process.env.AUTH0_CLIENT_SECRET
+    })
+  }
+
+  authenticate[Symbol.for('skip-override')] = true
+  fastify.register(authenticate)
+
+  // Setup auth0 protected route
+  fastify.register(async function protectedRoute(fastify) {
```

Comment:
same as before, is there a reason to register routes this way?

Reply:
Auth protected routes seem to need to be registered via `fastify.register`? The readme propose the same syntax and registering without it seems to skip the necessary `preValidation` step.

Reply:
have you tried?

Reply:
you're jumping to conclusions too quickly. the readme isn't wrong, but it's showing a workaround that's unnecessary and looks more like a hack than anything else. please look into it more closely

Reply:
I could make it work with (another, smaller) workaround:

```js
  server.get('/protected', { preHandler: (...args) => server.authenticate(...args) }, (req, reply) => {
    reply.send({ route: 'Protected route' })
  })
```

`server.authenticate` is still `undefined` when immediately referenced from a route declaration.

When the route declaration in executed asynchronously inside a an async `fastify.register` callback or (in this case) from within a hook callback, then Fastify has been already decorated with the relecant `authenticate` method.

Do you think it's worth using such workaround?

Reply:
Found a better approach awaiting the `.register` call.